### PR TITLE
Use the OpenAstronomy GitHub workflows

### PR DESCRIPTION
**Pull request by Cadair**

### CODE REVIEW

**Comment by Cadair** [comment]
rebased.

**Comment by pllim** [comment]
Is it possible to not change the displayed job names (like having "/ tests /" or "/ allowed_failures /" so I don't have to update the branch protection rules?

**Comment by Cadair** [comment]
I don't think so, at least not without modifying the templates.

**Comment by pllim** [comment]
Can we modify the templates? That text seems redundant and makes the CI name even longer.

**Comment by Cadair** [comment]
Hmm, I don't actually think it's in the templates from a quick look it might be a thing actions does when importing a workflow. I shall ask @ConorMacBride

**Review by astrofrog — APPROVED** [review]
From my point of view this looks good if CI passes. While it would be nice to be able to not change the display names, if it's going to require a lot of work then I think updating the branch protection shouldn't take that much time, right?

**Comment by pllim** [comment]
> updating the branch protection shouldn't take that much time, right?

Would be a pain if you have a lot of other repos though... Is it not possible at all?

**Comment by Cadair** [comment]
As far as I can tell it's not us. In the workflows we just set the name to be what is coming after the last slash: https://github.com/OpenAstronomy/github-actions-workflows/blob/main/.github/workflows/tox.yml#L124

**Comment by pllim** [comment]
Where is this coming from?

https://github.com/OpenAstronomy/github-actions-workflows/blob/2a7b8a89aa6eaf0cb6b87159e8913e99f819f4bf/.github/workflows/tox.yml#L130

**Comment by pllim** [comment]
Hmm for whatever reason, it is prepending the name of the workflow in this repo and the name of the job in this repo before showing the name in your workflow. 🤔  You sure you didn't set it up like that somewhere? I cannot follow the upstream script. Too many redirections.

**Comment by Cadair** [comment]
> Where is this coming from?
> 
> https://github.com/OpenAstronomy/github-actions-workflows/blob/2a7b8a89aa6eaf0cb6b87159e8913e99f819f4bf/.github/workflows/tox.yml#L130

It gets generated in the "Load tox environments" step you can see what gets passed through as input in the output of that job: https://github.com/astropy/astropy/actions/runs/3172864970/jobs/5167825572#step:6:88

I really think it's GH doing it, as the input to the matrix is just the bit after the slash and we aren't modifying it anywhere.

**Comment by pllim** [comment]
Maybe this is an unavoidable one-time cost to transition then. How unfortunate.

**Comment by pllim** [comment]
Let's wait till after the Cloud FITS PR is merged first. Poor Geert.

**Comment by ConorMacBride** [comment]
> Hmm, I don't actually think it's in the templates from a quick look it might be a thing actions does when importing a workflow. I shall ask @ConorMacBride

Sorry for the delay. As far as I can tell, the `X / Y / Z` is just GitHub's way of showing that job `Z` is a job specified by the reusable workflow called by job `Y` within workflow file `X`. I don't see an obvious way to remove the `X / Y / `. In the reusable workflow we only define the job [like this](https://github.com/OpenAstronomy/github-actions-workflows/blob/2a7b8a89aa6eaf0cb6b87159e8913e99f819f4bf/.github/workflows/tox.yml#L124) which is just the specified `name:` (which defaults to the name of the tox environment) with the GitHub runner/OS name added to the end.

I'm not familiar with configuring required checks, but [this example in the GitHub docs](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#example) seems to suggest that you only need to specify the name of the job, i.e., `Z` only? If that is the case, we could update the reusable tox workflow to not append the runner name when a job name has been specified by `name:` (as that's what you are doing in this PR rather than defaulting to using the tox env name). That way it should never be necessary to change the branch protection rules.

**Comment by astrofrog** [comment]
Ok so I think the best plan here is to wait until after the 5.2 feature freeze and before the Black transition period when open PRs should be at a minimum - as merging this and changing the branch protection rules will likely require other PRs to rebase to satisfy the required CI.

**Comment by Cadair** [comment]
Sorry for the delay, but I have now rebased this, hopefully it all works.

**Review by pllim — COMMENTED** [review]
Failure is unrelated. The name change still annoying but I guess I'll get over it because I like `+86 −185`. Thanks!

**Comment by astrofrog** [comment]
I'm going to go ahead and merge this and will update the branch protection rules

**Comment by pllim** [comment]
@astrofrog , did you manage to update the new rules or do you still need help?

**Comment by pllim** [comment]
Update: I added the new test names to the branch protection rules. We can tweak more as needed.

File: .github/workflows/ci_cron_weekly.yml
Comment (pllim) [inline .github/workflows/ci_cron_weekly.yml:35]: This don't go together. It is a syntax error. If you check the jobs for this PR, weekly cron jobs were missing. This was merged prematurely. See https://github.com/astropy/astropy/pull/14070

**Comment by pllim** [comment]
This never made it to v5.2.x and given https://github.com/astropy/astropy/pull/14077 , probably easier not to backport, so I updated the milestone.

**Pull request merged**
